docs/integration_pipeline_controller.md

```diff
@@ -0,0 +1,37 @@
+<div align="center"><h1>IntegrationPipeline Controller</h1></div>
+
+```mermaid
+
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+  classDef Red fill:#FF9999;
+  classDef Amber fill:#FFDEAD;
+  classDef Green fill:#BDFFA4;
+
+   %% Node definitions
+  predicate((PREDICATE: <br>integration pipeline <br> reconciliation))
+  get_pipeline_run[Start pipeline run]
+  get_component[Get component]
+  get_application[Get application]
+  call_adapter[Call adapter with app, <br> component & pipeline run]
```

Comment:
The adapter holds the vast majority (or even all) of the business logic, so I think this node should be replaced with nodes explaining what happens in the adapter, like:

- Report the status of the integration pipeline run if the Snapshot was created for Pull requests
- Check if Snapshot passed all tests and update its status accordingly
- Check if the Snapshot needs to be superseded with a composite Snapshot
- Make sure that the ephemeral environment is cleaned up if the testing finished